### PR TITLE
Tackle git maven integration

### PR DESCRIPTION
**Pull request by rromannissen**
Application source code and binaries are typically stored and managed using corporate repositories. Most projects from the Tackle toolkit require access to source code to perform their function, and manually downloading source and/or binaries doesn't work well at scale. This enhancement proposes the integration of Tackle with the most widespread source and binaries repositories to automate the task of gathering source and binaries for them to be fed to the Tackle tools that might require them. Since credentials are required to access these repositories in most cases, a big part of this enhancement is dedicated to safely store and manage them.

### CODE REVIEW

File: enhancements/tackle/tackle-git-maven-integration/README.md [hunk 1]
```diff
@@ -0,0 +1,714 @@
...
+
+### Goals
+
+- Enhance the Application entity to store information about source and binary repositories.
+- Provide a way to store and manage credentials safely.
+- Add an administration view that can accommodate application wide configurations and sensitive data management.
+
+### Non-Goals
+
+- Provide clone/checkout/download logic for addons to get access to the source or binaries. Each addon will be responsible of gathering source or binaries by itself using the credentials provided by Tackle Hub.
```
Comment (PhilipCattanach) [inline enhancements/tackle/tackle-git-maven-integration/README.md:56]: This implies that the credentials will be shared with other add-ons that wish to download the source/artifacts. Wouldn't a single central service that provides that download capability be more secure and appropriate?

File: enhancements/tackle/tackle-git-maven-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+A technical lead for the migration project that can create and modify applications and information related to it. The Architects don’t need to have access to sensitive information, but can consume it. Example: Associate an existing credential to access the repository of a given application.
+
+#### Migrator
+
+A developer that should be allowed to run assessments and analysis, but not to create or modify applications in the portfolio.
+
+
+#### Organization Administrator
+
+A member of the operations or security team of the organization undergoing the adoption/migration initiative. The Organization Administrator manages the credential when there are security constraints that would prevent the Administrator from managing that information (For example when the Administrator belongs to the adoption team and that has been outsourced to a third party company).
```
Comment (PhilipCattanach) [inline enhancements/tackle/tackle-git-maven-integration/README.md:78]: The Administrator role will have update access to the credential entities. I can't see how we can differentiate between an external or organization Administrator for a system perspective. They either have update access to credentials or they don't. From a security perspective it must be very desirable for all credentials management to be done by internal resources.

File: enhancements/tackle/tackle-git-maven-integration/README.md [hunk 3]
```diff
@@ -0,0 +1,714 @@
...
+
+
+
+### User Stories
```
Comment (PhilipCattanach) [inline enhancements/tackle/tackle-git-maven-integration/README.md:82]: General comment. All of the User Stories need acceptance criteria.

File: enhancements/tackle/tackle-git-maven-integration/README.md [hunk 4]
```diff
@@ -0,0 +1,714 @@
...
+
+*As an Administrator I want to be able to manage global Git configuration.*
+
+##### GC002
+
+*As an Administrator I want to be able to define the default behavior for whether new changes in a repository are acquired via a pull or by performing a full clone on each analysis.*
+
+##### GC003
+
+*As an Administrator I want to be able to configure if insecure Git repositories can be consumed.*
```
Comment (PhilipCattanach) [inline enhancements/tackle/tackle-git-maven-integration/README.md:141]: The term 'insecure'. 
Does this mean accessed without SSL or with a self signed SSL certificate?
Perhaps we could have a Terminology section at the beginning of the document that covers this?

File: enhancements/tackle/tackle-git-maven-integration/README.md [hunk 5]
```diff
@@ -0,0 +1,714 @@
...
+
+*As an Administrator I want to be able to configure the default behavior for forced updates for artifacts and dependencies on each analysis.*
+
+##### MC003
+
+*As an Administrator I want to be able to configure if insecure Artifact Repositories can be consumed.*
+
+##### MC004
+
+*As an Administrator I want to be able to know the size of the local artifact repository.*
```
Comment (PhilipCattanach) [inline enhancements/tackle/tackle-git-maven-integration/README.md:193]: Terminology: local artifact repository. I guess this must be a shared PV that all of the analysis pods will have Read Write Many access to?

File: enhancements/tackle/tackle-git-maven-integration/README.md (hunk 1 above)
Comment (jortel) [reply enhancements/tackle/tackle-git-maven-integration/README.md:56]: The credentials would not be shared with the actual addon only used by the addon _adapter_ which is part of tackle and trusted.

File: enhancements/tackle/tackle-git-maven-integration/README.md (hunk 5 above)
Comment (jortel) [reply enhancements/tackle/tackle-git-maven-integration/README.md:187]: Yes.

**Pull request merged**